pom.xml

```diff
@@ -341,6 +341,8 @@
                         <!-- Warn if an argument list is modified to match the receiver. -->
                         <arg>-Ywarn-adapted-args</arg>
                         <arg>-Ywarn-dead-code</arg>
+                        <!-- Fail on compiler warnings -->
+                        <!--<arg>-Xfatal-warnings</arg>-->
```

Comment:
Why do we add those comments ?

Reply:
I was on the fence. I left them in as a todo, as the intention is eventually to enable fail-on-warnings mode, but we have to whack down the warnings first.

Aside from the "todo", leaving them here just saves a little bit of research time when I'd have to look up how to enable it again.

But I can take it out if you prefer.

Reply:
I decided to leave them there, but add an actual TODO stating when the option should be enabled.